docs/configuration.md

```diff
@@ -358,25 +744,112 @@ http:
+    </td>
+  </tr>
+    <tr>
+    <td>
+      <code>prefix</code>
+    </td>
+    <td>
+      no
+    </td>
+    <td>
+If the server does not run at the root path use this value to specify the
+prefix. The root path is the section before <code>v2</code>. It
+should have both preceding and trailing slashes, for example <code>/path/</code>.
+    </td>
+  </tr>
+  <tr>
+    <td>
+      <code>secret</code>
+    </td>
+    <td>
+      yes
+    </td>
+    <td>
+A random piece of data. This is used to sign state that may be stored with the
+client to protect against tampering. For production environments you should generate a
+random piece of data using a cryptographically secure random generator.
+    </td>
+  </tr>
+</table>
 
-- addr: **Required** - The HOST:PORT for which the server should accept connections.
-- prefix: **Optional** - If the server will not run at the root path, this should specify the prefix (the part of the path before ```v2```). It should have both preceding and trailing slashes.
-- secret: A random piece of data. It is used to sign state that may be stored with the client to protect against tampering. For production use you should generate a random piece of data using a cryptographically secure random generator.
 
 ### tls
 
-The tls option within http is **optional** and allows you to configure SSL for the server. If you already have a server such as Nginx or Apache running on the same host as the registry, you may prefer to configure SSL termination there and proxy connections to the registry server.
+The `tls` struct within `http` is **optional**. Use this to configure SSL
+for the server. If you already have a server such as Nginx or Apache running on
+the same host as the registry, you may prefer to configure SSL termination there
+and proxy connections to the registry server.
+
+<table>
+  <tr>
+    <th>Parameter</th>
+    <th>Required</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>
+      <code>certificate</code>
+    </td>
+    <td>
+      yes
+    </td>
+    <td>
+       Absolute path to x509 cert file
+    </td>
+  </tr>
+    <tr>
+    <td>
+      <code>key</code>
+    </td>
+    <td>
+      yes
+    </td>
+    <td>
+      Absolute path to x509 private key file.
+    </td>
+  </tr>
+  <tr>
+    <td>
+      <code>clientcas</code>
+    </td>
+    <td>
+      no
+    </td>
+    <td>
+      An array of absolute paths to a x509 CA file
+    </td>
+  </tr>  
+</table>
 
-- certificate: **Required** - Absolute path to x509 cert file
-- key: **Required** - Absolute path to x509 private key file
-- clientcas: **Optional** - An array of absolute paths to a x509 CA file
 
 ### debug
 
-The debug option is **optional** and allows you to configure a debug server that can be helpful in diagnosing problems. It is of most use to contributers to the distribution repository and should generally be disabled in production deployments.
+The `debug` option is **optional** . Use it to configure a debug server that can
```

Comment:
The debug server really exposes internal runtime state via `pprof` and `expvar`. Not sure how much detail we should have about that here. Maybe just take note/